account_payment_term_partner_holiday/models/res_partner.py

```diff
@@ -19,28 +21,28 @@ class ResPartner(models.Model):
     def is_date_in_holiday(self, date):
         if isinstance(date, str):
             date = fields.Date.from_string(date)
-        res = self.holiday_ids.filtered(
-            lambda x: int(x.month_from) <= date.month and int(x.month_to) >= date.month
-            and int(x.day_from) <= date.day and int(x.day_to) >= date.day
-        )
-        if bool(res):
-            res = res[0]
-            res_date_to = False
-            day_to = int(res.day_to)
-            while not res_date_to:
-                try:
-                    res_date_to = fields.Date.from_string(
-                        "%s-%s-%s" % (date.year, res.month_to, day_to)
-                    )
-                except:
-                    day_to -= 1
-            return [
-                fields.Date.from_string(
-                    "%s-%s-%s" % (date.year, res.month_from, res.day_from)
-                ), res_date_to
-            ]
+        for holiday in self.commercial_partner_id.holiday_ids:
+            holiday_start_date = self._generate_field_date(
+                date.year, int(holiday.month_from), int(holiday.day_from)
+            )
+            # If the user selected a date of the next year (e.g. vacations from
+            # December to January), compute the correct date
+            year_end = date.year
+            if holiday.month_to < holiday.month_from:
```

Comment:
This should be forbidden in the creation, not covered here. We already talked that the constraint should be `month_from` always < `month_to`.

Reply:
Is the table ordered by day_from and month_from by the way? It should.